tst/EnergyPlus/unit/EMSManager.unit.cc

```diff
@@ -1558,5 +1571,39 @@ TEST_F(EnergyPlusFixture, EMSManager_TestOANodeAsActuators)
     EXPECT_TRUE(Node(1).IsLocalNode);
     EXPECT_FALSE(Node(2).IsLocalNode);
     EXPECT_TRUE(Node(3).IsLocalNode);
-
+}
+TEST_F(EnergyPlusFixture, EMSManager_TestWindowShadingControlExteriorScreenOption2b)
+{
+    // #7586
+    DataSurfaces::Surface.allocate(2);
+    DataSurfaces::SurfaceWindow.allocate(2);
+    DataHeatBalance::Construct.allocate(1);
+    DataSurfaces::WindowShadingControl.allocate(2);
+    DataSurfaces::Surface(1).Name = "Surface1";
+    DataSurfaces::Surface(2).Name = "Surface2";
+    DataSurfaces::Surface(1).Class = DataSurfaces::SurfaceClass_Window;
+    DataSurfaces::Surface(2).Class = DataSurfaces::SurfaceClass_Window;
+    DataSurfaces::Surface(1).ExtBoundCond = DataSurfaces::ExternalEnvironment;
+    DataSurfaces::Surface(2).ExtBoundCond = DataSurfaces::ExternalEnvironment;
+    DataSurfaces::Surface(1).WindowShadingControlPtr = 1;
+    DataSurfaces::Surface(2).WindowShadingControlPtr = 2;
+    DataSurfaces::Surface(1).HasShadeControl = true;
+    DataSurfaces::Surface(2).HasShadeControl = true;
+
+    DataSurfaces::SurfaceWindow(1).HasShadeOrBlindLayer = false;
+    DataSurfaces::SurfaceWindow(2).HasShadeOrBlindLayer = false;
+    DataSurfaces::SurfaceWindow(1).ShadedConstruction = 1;
+    DataSurfaces::SurfaceWindow(2).ShadedConstruction = 1;
+
+    DataHeatBalance::Construct(1).Name = "Construction1";
+
+    DataSurfaces::WindowShadingControl(1).ShadingType = 0;
+    DataSurfaces::WindowShadingControl(2).ShadingType = DataSurfaces::WSC_ST_ExteriorScreen;
+
+    DataSurfaces::TotSurfaces = 2;
+
+    SetupWindowShadingControlActuators();
+
+    EXPECT_FALSE(DataSurfaces::SurfaceWindow(2).ShadingFlagEMSOn);
+    EXPECT_EQ(DataSurfaces::SurfaceWindow(2).ShadingFlagEMSValue,0);
```

Comment:
I'm not sure these lines actually reflect the test.  These values on the window data structure are default initialized to false and zero, regardless of whether an EMS actuator was actually set up.  I think what you'd want to do is make sure an EMS actuator was instantiated in the actuator list with these names and control types.  I'm not sure if there is an example of that in the unit tests already or if there is a helper function in the EMS code that would make it easy, but if you need assistance let me know.

Reply:
@Myoldmopar You are right. The unit test ensure that EMS actuator is instantiated. I have a test file available in DevelopmentSupport\EnergyPlusDevSupport\DefectFiles\7000s\7586 as 7586b.idf to check if the EMS actuator is working or not.

Reply:
@Myoldmopar I added a section in the unit test to ensure that the EMS actuator overrides the ShadingFlag value.